### PR TITLE
Add ALL_MODULES_EXCEPT_MAIN token for compiled mode

### PR DESCRIPTION
**Pull request by pfitzseb**

### CODE REVIEW

File: src/debugger_requests.jl [hunk 1]
```diff
@@ -128,6 +128,15 @@ function set_compiled_functions_modules!(items::Vector{String})
 
     # user wants these compiled:
     for acc in items
+        if acc == "ALL_MODULES_EXCEPT_MAIN"
+            for mod in values(Base.loaded_modules)
```
Comment (MariusDrulea) [inline src/debugger_requests.jl:132]: When does this code run? After debugger launches and after the ```my_script.jl``` is loaded? ```my_script.jl``` is the script the user wants to debug

Comment (MariusDrulea) [inline src/debugger_requests.jl:132]: I noticed in some of my scripts that the _number of modules increases while running_. That means these additional modules will not be set to compiled.

Comment (MariusDrulea) [inline src/debugger_requests.jl:132]: Question: does the loaded modules contain all submodules as well? The submodules shall also be set to ```compiled```, right?

Comment (pfitzseb) [reply src/debugger_requests.jl:132]: This ends up being run on each debugger step.

**Review by MariusDrulea — APPROVED** [review]
looks good

**Review by MariusDrulea — APPROVED** [review]

**Review by davidanthoff — APPROVED** [review]

**Pull request merged**